test/utils/DataUtils.sol

```diff
@@ -42,6 +33,10 @@ contract DataUtils is
 
     address internal _admin;
     address internal _userOne;
+    address internal _userTwo;
+    address internal _userThree;
+    address internal _liquidityProvider;
+    address[] internal _users;
 
     function setupIporProtocolForUsdt() public returns (IporProtocol memory iporProtocol) {
         MockTestnetToken asset = getTokenUsdt();
```

Comment:
![Screen Shot 2023-01-24 at 9 17 36 AM](https://user-images.githubusercontent.com/103587277/214242417-7bb58324-daa3-46c8-987e-b78032a41a38.png)

I do it like this because the return value expects a `MockTestnetToken`

Reply:
```
 function setupIporProtocolForUsdt() public returns (IporProtocol memory iporProtocol) {
        address asset = address(getTokenUsdt());
        IpToken ipToken = getIpTokenUsdt(asset);
        ItfStanley stanley = getItfStanleyUsdt(asset);
        MiltonStorage miltonStorage = getMiltonStorage();

        address[] memory tokenAddresses = new address[](1);
        tokenAddresses[0] = asset;

        ItfIporOracle iporOracle = getIporOracleAssets(_userOne, tokenAddresses, 1, 1, 1);

        MockIporWeighted iporWeighted = _prepareIporWeighted(address(iporOracle));

        iporOracle.setIporAlgorithmFacade(address(iporWeighted));

        MockSpreadModel miltonSpreadModel = prepareMockSpreadModel(0, 0, 0, 0);

        ItfMilton milton = getItfMiltonUsdt(
            asset, address(iporOracle), address(miltonStorage), address(miltonSpreadModel), address(stanley)
        );

        ItfJoseph joseph = getItfJosephUsdt(
            asset, address(ipToken), address(milton), address(miltonStorage), address(stanley)
        );

        prepareIpToken(ipToken, address(joseph));
        prepareJoseph(joseph);
        prepareMilton(milton, address(joseph), address(stanley));

        iporOracle.setIporAlgorithmFacade(address(iporWeighted));
        stanley.setMilton(address(milton));

        return IporProtocol(MockTestnetToken(asset), ipToken, stanley, miltonStorage, milton, joseph);
    }
```

I think I will cast only once at the end to MockTestnetToken